design-documents/graph-ql/coverage/b2b/purchase-order.graphqls

```diff
@@ -0,0 +1,303 @@
+    updatePurchaseOrderApprovalRule(input: UpdatePurchaseOrderApprovalRuleInput!): UpdatePurchaseOrderApprovalRuleOutput @doc(description: "Update existing purchase order approval rule")
+    deletePurchaseOrderApprovalRule(input: DeletePurchaseOrderApprovalRuleInput!): DeletePurchaseOrderApprovalRuleOutput @doc(description: "Delete existing purchase order approval rule")
+}
+
+type CreatePurchaseOrderApprovalRuleOutput {
+    approval_rule: PurchaseOrderApprovalRule @doc(description: "Created purchase order approval rule")
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+type UpdatePurchaseOrderApprovalRuleOutput {
+    approval_rule: PurchaseOrderApprovalRule @doc(description: "Updated purchase order approval rule")
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+type DeletePurchaseOrderApprovalRuleOutput {
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+input CreatePurchaseOrderApprovalRuleInput {
+    approval_rule: PurchaseOrderApprovalRuleInput! @doc(description: "Purchase order approval rule data")
+}
+
+input UpdatePurchaseOrderApprovalRuleInput {
+    approval_rule_uid: ID! @doc(description: "Purchase order approval rule ID")
+    approval_rule: PurchaseOrderApprovalRuleInput! @doc(description: "Purchase order approval rule data")
+}
+
+input DeletePurchaseOrderApprovalRuleInput {
+    approval_rule_uid: ID! @doc(description: "Purchase order approval rule ID")
+}
+
+input PurchaseOrderApprovalRuleInput {
+    name: String! @doc(description: "Purchase order approval rule name")
+    description: String @doc(description: "Purchase order approval rule description")
+    applies_to: [ID]! @doc(description: "A list of B2B user roles to which this purchase order approval rule should be applied. In case when empty array is provided, the rule will be applied to all user roles in the system, including those created in the future")
+    type: PurchaseOrderApprovalRuleType! @doc(description: "Purchase order approval rule type")
+    condition: CreatePurchaseOrderApprovalRuleConditionInput! @doc(description: "Purchase order approval rule condition")
+    requires_approval_from: [ID!]! @doc(description: "A list of B2B user roles that can approve this purchase order approval rule")
+}
+
+input CreatePurchaseOrderApprovalRuleConditionInput {
+    operator: PurchaseOrderApprovalRuleConditionOperator! @doc(description: "Purchase order approval rule condition operator")
+    amount: CreatePurchaseOrderApprovalRuleConditionAmountInput @doc(description: "Purchase order approval rule condition ammount. Is mutually exclusive with condition quantity")
```

Comment:
In case of Number of SKUs option, we don't have to specify currency and have quantity instead of amount. Can we introduce subtypes here?

Reply:
Unfortunately no, because inputs do not support polymorphism. 
It can be done this way, or via separate mutations, which will have independent input structures.